pinot-core/src/main/java/org/apache/pinot/core/data/table/IndexedTable.java

```diff
@@ -38,6 +38,7 @@
 @SuppressWarnings({"rawtypes", "unchecked"})
 public abstract class IndexedTable extends BaseTable {
   protected final Map<Key, Record> _lookupMap;
+  protected final boolean _hasFinalInput;
```

Comment:
Since this IndexedTable has multiple aggregations, do you need this `_hasFinalInput` also to be per agg function or it's just an indicate ?
E.g. `SELECT distinctCount(pk), avg(other_column) from myTable`

Reply:
In V1 engine we don't have a way to make per function hint. We can make it more smart in V2 which can carry more information.